src/components/admin/admin.service.ts

```diff
@@ -1,151 +1,141 @@
-    );
-    wait.push(
-      session.run('CREATE CONSTRAINT ON (area:Area) ASSERT exists(area.id)')
-    );
-    wait.push(
-      session.run('CREATE CONSTRAINT ON (area:Area) ASSERT exists(area.name)')
-    );
-    wait.push(
-      session.run('CREATE CONSTRAINT ON (area:Area) ASSERT exists(area.active)')
-    );
-
-    await Promise.all(wait);
-    session.close();
+export class AdminService implements OnApplicationBootstrap {
+  constructor(
+    private readonly db: DatabaseService,
+    private readonly config: ConfigService,
+    private readonly userService: UserService
+  ) {}
+
+  async onApplicationBootstrap(): Promise<void> {
+    if (!(await this.rootAdminSecurityGroupExists())) {
+      await this.createRootAdminSecurityGroup();
+    }
 
-    response.success = true;
+    if (!(await this.doesRootAdminUserAlreadyExist())) {
+      await this.createRootAdminUser();
+    }
 
-    return response;
+    await this.mergeRootAdminUserToSecurityGroup();
   }
 
-  async loadFakeData(): Promise<AdminOutputDto> {
-    const response = new AdminOutputDto();
-
-    const totalUsers = 50;
-    const totalOrgs = 50;
-
-    // ORGS
-    for (let i = 0; i < totalOrgs; i++) {
-      // this.orgService.create({
-      //   name: 'org_' + generate(),
-      // });
+  async rootAdminSecurityGroupExists(): Promise<boolean> {
+    const result = await this.db
+      .query()
+      .match([[node('sg', 'RootSecurityGroup')]])
+      .return('sg')
+      .first();
+
+    if (result) {
+      return true;
+    } else {
+      return false;
     }
+  }
 
-    // USERS
-    for (let i = 0; i < totalUsers; i++) {
-      // const s = 'user_' + generate();
-      // await this.userService.create(
-      //   {
-      //     displayFirstName: s,
-      //     displayLastName: s,
-      //     email: s,
-      //     realFirstName: s,
-      //     realLastName: s,
-      //   },
-      //   'token - replace me later',
-      // );
+  async doesRootAdminUserAlreadyExist(): Promise<boolean> {
+    const result = await this.db
+      .query()
+      .match([
+        [
+          node('user', 'User'),
+          relation('out', '', 'email', {
+            active: true,
+          }),
+          node('email', 'EmailAddress', {
+            value: this.config.rootAdmin.email,
+          }),
+        ],
+      ])
+      .return('user')
+      .first();
+
+    if (result) {
+      return true;
+    } else {
+      return false;
     }
-
-    response.success = true;
-
-    return response;
   }
 
-  async consistencyCheck(): Promise<AdminOutputDto> {
-    const response = new AdminOutputDto();
-    let totalNodes = 0;
-    const session = this.db.driver.session();
-    const result = await session.run('MATCH (n) RETURN count(n) as count', {});
-    totalNodes = result.records[0].get('count');
-    for (let i = 0; i < totalNodes; i++) {
-      // todo
+  async createRootAdminSecurityGroup(): Promise<boolean> {
```

Comment:
We don't want to return booleans to indicate success. Throw an error if there is a failure.

Reply:
Done

Reply:
```suggestion
  async createRootAdminSecurityGroup(): Promise<void> {
```